src/os/bluestore/BlueStore.cc

```diff
@@ -7628,8 +7631,12 @@ bool BlueStore::test_mount_in_use()
 int BlueStore::_minimal_open_bluefs(bool create)
 {
   int r;
-  bluefs = new BlueFS(cct);
 
+  if (cct->_conf->bluefs_enable_cache) {
+    bluefs = new BlueFS(cct, bluefscache);
+  } else {
+    bluefs = new BlueFS(cct);
+  }
```

Comment:
Can't it simply be:
`bluefs = new BlueFS(cct, bluefscache);` ?
If bluefscache was not initialized it will be nullptr, which you get anyway from BlueFS::BlueFS() constructor.

Reply:
Updated